### PR TITLE
Books sign up

### PR DESCRIPTION
**Pull request by eledwinn**
## Summary

1. Sign up form created and connected with API

## Screenshot

sign-up

## Trello Card
https://trello.com/c/RU04KefK/25-sign-up

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "Environment": {
+      "wBooksApiUrl": "https://wbooks-api-stage.herokuapp.com/api/v1/"
+    }
+  },
+  "production": {
+    "Environment": {
+      "wBooksApiUrl": "http://polls.apiblueprint.org/api/v1/"
+    }
+  }
+}
```
Comment (FrankIglesias) [inline config/config.json]: add eol

File: src/scss/button.scss [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
 .btn-primary {
   background: $blue;
-  color: $white;
+}
+
+.btn-secondary {
+  background: $green;
+}
+
+.btn:disabled {
```
Comment (FrankIglesias) [inline src/scss/button.scss]: use `&`

File: src/sign-up/sign-up.component.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+angular.module('signUp').component('signUp', {
+  templateUrl: 'sign-up/sign-up.template.html',
+  bindings: { books: '<' },
+  controller: ['$state', 'Users', function BooksListController($state, Users) {
+    this.passwordValidatorPattern = new RegExp('^(?=.*[A-z])(?=.*[0-9]).{8,52}$');
```
Comment (FrankIglesias) [inline src/sign-up/sign-up.component.js]: move this regex to a constant file

File: src/sign-up/sign-up.component.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          password: this.user.password,
+          password_confirmation: this.user.password,
+          first_name: this.user.firstName,
+          last_name: this.user.lastName,
+          locale: 'es'
+        }
+      };
+      this.loading = true;
+      Users.create(user).then(() => {
+        // $state.go('logIn'); Available in next card.
```
Comment (FrankIglesias) [inline src/sign-up/sign-up.component.js]: remove comment

File: src/sign-up/sign-up.template.pug [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          | Nombre
+        input#firstName.form-control(type="text" name="firstName" ng-model="$ctrl.user.firstName" ng-pattern="/^[a-zA-Z\s]*$/" required)
+        p.form-validator-msg(ng-if="signUpForm.firstName.$error.required && signUpForm.firstName.$touched")
+          | Ingrese su nombre
+        p.form-validator-msg(ng-if="signUpForm.firstName.$error.pattern")
+          | Sólo se pueden ingresar letras y espacios
+      .form-group
+        label.form-label(for="lastName")
+          | Apellido
+        input#lastName.form-control(type="text" name="lastName" ng-model="$ctrl.user.lastName" ng-pattern="/^[a-zA-Z\s]*$/" required)
```
Comment (FrankIglesias) [inline src/sign-up/sign-up.template.pug]: this regex can be done with a variable

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: src/app.js [hunk 6]
```diff
@@ -6,31 +6,35 @@ angular.module('wBooksApp', [
...
+  }, {
+    name: 'bookDetail',
+    url: '/books/{id}',
     component: 'bookDetail',
     resolve: {
       book: (Books, $stateParams) => {
         return Books.get().then(data => {
-          return data.find(book => book.id == $stateParams.id);
-        })
+          return data.find(book => book.id === +$stateParams.id);
```
Comment (FrankIglesias) [inline src/app.js:28]: why this plus symbol?

File: src/books/books.component.js [hunk 7]
```diff
@@ -1,24 +1,23 @@
...
-        display: "Nombre",
-        value: "title"
-      }, {
-        display: "Autor",
-        value: "author"
-      }];
+angular.module('booksList').component('booksList', {
+  templateUrl: 'books/books.template.html',
+  bindings: { books: '<' },
+  controller: function BooksListController() {
```
Comment (FrankIglesias) [inline src/books/books.component.js]: remove the alias to the function if you are not going to use it

File: src/books/books.component.js [hunk 8]
```diff
@@ -1,24 +1,23 @@
...
 
-      this.search = () => {
-        this.filter = {};
-        if (this.filterField && this.filterField.length > 0 && this.filterCriteria && this.filterCriteria.trim().length > 0) {
-          this.filter[this.filterField] = this.filterCriteria;
-        }
+    this.search = () => {
+      this.filter = {};
+      if (this.filterField && this.filterField.length > 0
+        && this.filterCriteria && this.filterCriteria.trim().length > 0) {
```
Comment (FrankIglesias) [inline src/books/books.component.js]: delete the `> 0`

**Review by FrankIglesias — APPROVED** [review]

**Review by FrankIglesias — CHANGES_REQUESTED** [review]
Correct the comments

File: src/app.js (hunk 6 above)
Comment (eledwinn) [reply src/app.js:28]: $stateParams.id is numeric string, with + symbol convert implicit to number to compare with book.id.

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**